control-plane/api-gateway/gatekeeper/deployment.go

```diff
@@ -75,8 +75,8 @@ func (g *Gatekeeper) upsertDeployment(ctx context.Context, gateway gwv1beta1.Gat
 	return nil
 }
 
-func (g *Gatekeeper) deleteDeployment(ctx context.Context, nsname types.NamespacedName) error {
-	err := g.Client.Delete(ctx, &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Name: nsname.Name, Namespace: nsname.Namespace}})
+func (g *Gatekeeper) deleteDeployment(ctx context.Context, gwName types.NamespacedName) error {
```

Comment:
These changes were just improving a variable name that was confusing to me coming into the project near the end